code/modules/mob/living/simple_animal/friendly/mouse.dm

```diff
@@ -95,12 +108,49 @@
 	src << "<span class='warning'>You are too small to pull anything.</span>"
 	return
 
+
+
+//Plays a sound.
+//This is triggered when a mob steps on an NPC mouse, or manually by a playermouse
+/mob/living/simple_animal/mouse/proc/squeak()
+	set name = "Squeak"
+	set category = "Abilities"
+	playsound(src, 'sound/effects/mousesqueek.ogg', 90, 1)
+
+
+
+
+//Plays a random selection of four sounds, at a low volume
+//This is triggered randomly periodically by any mouse, or manually
+/mob/living/simple_animal/mouse/proc/squeak_soft()
+	set name = "Soft Squeaking"
+	set category = "Abilities"
+
+	var/sound = pick(soft_squeaks)
+	while (sound == last_softsqueak)

```

Comment:
Why use a while() loop, and not do this:
```
var/list/new_squeaks = sound ? soft_squeaks - sound : soft_squeaks
sound = pick(new_squeaks)
```

Seems like it'd be more efficient that way?